chia/wallet/wallet_state_manager.py

```diff
@@ -275,17 +262,13 @@ async def create(
     def get_public_key_unhardened(self, index: uint32) -> G1Element:
         return master_sk_to_wallet_sk_unhardened(self.private_key, index).get_g1()
 
-    async def get_keys(self, puzzle_hash: bytes32) -> Optional[Tuple[G1Element, PrivateKey]]:
+    async def get_private_key(self, puzzle_hash: bytes32) -> PrivateKey:
         record = await self.puzzle_store.record_for_puzzle_hash(puzzle_hash)
         if record is None:
-            raise ValueError(f"No key for this puzzlehash {puzzle_hash})")
```

Comment:
so, this never returned `None`, so all the code you removed checking for none was dead code all along. Are you sure we don't do anything important in those snippets?

Reply:
We just raise (and log in one case which is not critical to drop the log imo) which this method does already.